misc/snapshotter/Dockerfile

```diff
@@ -1,12 +1,21 @@
 FROM alpine:3.17.0 AS sourcer
 
+ARG TARGETARCH
+
 RUN apk add --no-cache curl
 RUN apk add --no-cache --upgrade grep
-RUN export NYDUS_VERSION=$(curl --silent "https://api.github.com/repos/dragonflyoss/image-service/releases/latest" | grep -Po '"tag_name": "\K.*?(?=")') && \
-    wget https://github.com/dragonflyoss/image-service/releases/download/$NYDUS_VERSION/nydus-static-$NYDUS_VERSION-linux-amd64.tgz && \
+RUN unset NYDUS_VERSION && \
+    while [ -z $NYDUS_VERSION ]; do \
+        export NYDUS_VERSION=$(curl --silent "https://api.github.com/repos/dragonflyoss/image-service/releases/latest" | grep -Po '"tag_name": "\K.*?(?=")'); sleep 1; \
+    done && \
+    echo NYDUS_VERSION: $NYDUS_VERSION && \
+    if [ -z $NYDUS_VERSION ]; then echo Invalid NYDUS_VERSION && exit -1; fi && \
```

Comment:
This condition judgment conflicts with line 8, which means the loop should not end.

Reply:
Yes. Now, we limit the total retry times, so we still need this.